src/app/components/Modals/ClaimRewardModal.tsx

```diff
@@ -0,0 +1,74 @@
+import { Heading, Text } from "@babylonlabs-io/bbn-core-ui";
+import { PropsWithChildren, useEffect, useState } from "react";
+
+import { LoadingSmall } from "../Loading/Loading";
+
+import { ConfirmationModal } from "./ConfirmationModal";
+
+interface ConfirmationModalProps {
+  open: boolean;
+  onClose: () => void;
+  onSubmit: () => void;
+  receivingValue: string;
+  address: string;
+  getTransactionFee: () => Promise<number>;
+}
+
+export const ClaimRewardModal = ({
+  open,
+  onClose,
+  onSubmit,
+  receivingValue,
+  address,
+  getTransactionFee,
+}: PropsWithChildren<ConfirmationModalProps>) => {
+  const [transactionFee, setTransactionFee] = useState<number>(0);
+
+  useEffect(() => {
+    const fetchTransactionFee = async () => {
+      const fee = await getTransactionFee();
+      setTransactionFee(fee);
+    };
+    fetchTransactionFee();
+  }, [getTransactionFee]);
```

Comment:
Don't write domain logic inside render function directly. Encapsulate it into related service

Reply:
@jrwbabylonlab let's discuss this on our call today as well